lib/tree.rb

```diff
@@ -1,38 +1,60 @@
 class NoApplesError < StandardError; end
 
-class AppleTree
-  attr_#fill_in :height, :age, :apples, :alive
+RANDOM = Random.new;
```

Comment:
Though definitely not needed here. This is a great time for me to point out...
http://ruby-doc.org/stdlib-2.1.2/libdoc/securerandom/rdoc/SecureRandom.html

We use it quite often in a number of places. Ask me tomorrow & lets discuss.